macOS/scripts/archive.sh

```diff
@@ -277,10 +288,54 @@ staple_notarized_app() {
 }
 
 compress_app_and_dsym() {
-	echo "Compressing app and dSYMs ..."
+	echo "Creating tar.xz archive and compressing dSYMs ..."
 
-	ditto -c -k --keepParent "${app_path}" "${output_app_zip_path}"
+	# Change to workdir to avoid full path in tar archive
+	pushd "${workdir}" > /dev/null
+	
+	# Temporarily rename app bundle to include version for consistency in archive
+	local versioned_app_name="DuckDuckGo-${version_identifier}.app"
+	local original_app_name="${app_name}.app"
+	
+	if [[ "${original_app_name}" != "${versioned_app_name}" ]]; then
+		echo "Temporarily renaming app bundle for archive: ${versioned_app_name}"
+		mv "${original_app_name}" "${versioned_app_name}"
```

Comment:
Please use `mv -f` – it’s a common practice in scripts, to avoid asking for permission on shells that have this configured.